lib/src/cli/deployapps/baas_client.dart

```diff
@@ -0,0 +1,294 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+
+class BaasClient {
+  static const String _confirmFuncSource = '''exports = ({ token, tokenId, username }) => {
+    // process the confirm token, tokenId and username
+    if (username.includes("realm_tests_do_autoverify")) {
+      return { status: 'success' }
+    }
+    // do not confirm the user
+    return { status: 'fail' };
+  };''';
+
+  static const String _resetFuncSource = '''exports = ({ token, tokenId, username, password }) => {
+    // process the reset token, tokenId, username and password
+    if (password.includes("realm_tests_do_reset")) {
+      return { status: 'success' };
+    }
+    // will not reset the password
+    return { status: 'fail' };
+  };''';
+
+  final String _baseUrl;
+  final String? _clusterName;
+  final String _appSuffix;
+  final Map<String, String> _headers;
+
+  late String _groupId;
+
+  BaasClient._(String baseUrl, [this._clusterName])
+      : _baseUrl = '$baseUrl/api/admin/v3.0',
+        _headers = <String, String>{'Accept': 'application/json'},
+        _appSuffix = '-$_clusterName';
+
+  static Future<BaasClient> docker(String baseUrl) async {
+    final result = BaasClient._(baseUrl);
+
+    await result._authenticate('local-userpass', '{ "username": "unique_user@domain.com", "password": "password" }');
+
+    dynamic groupDoc = await result._get('auth/profile');
+    result._groupId = (groupDoc['roles'] as List<dynamic>)[0]['group_id'] as String;
+
+    print('Current GroupID ${result._groupId}');
```

Comment:
Consider avoiding print directly and use logger instead (see metrics)

Reply:
I see it is a repeated issue across other commands as well. Lets consider this separately